lib/edoc/src/edoc.erl

```diff
@@ -60,23 +60,71 @@
 	 read_comments/1, read_comments/2,
 	 read_source/1, read_source/2]).
 
--compile({no_auto_import,[error/1]}).
+-export_type([module_meta/0,
+	      env/0,
+	      comment/0,
+	      entry/0,
+	      entry_data/0,
+	      tag/0,
+	      edoc_module/0]).
 
 -include("edoc.hrl").
 
+-type module_meta() :: #module{name :: [] | atom(),
+			       parameters :: none | [atom()],
+			       functions :: ordset(function_name()),
+			       exports :: ordset(function_name()),
+			       attributes :: ordset({atom(), term()}),
+			       records :: [{atom(), [{atom(), term()}]}],
+			       encoding :: epp:source_encoding()}.
+%% Module information.
+
+-type env() :: #env{}.
+%% Environment information needed by EDoc for generating references.
+
+-type comment() :: #comment{line :: integer(),
+			    text :: string()}.
+%% Simplified comment data.
+
+
+-type entry() :: #entry{name :: function_name() | atom(),
+			args :: [atom()],
+			line :: integer(),
+			export :: boolean(),
+			data :: entry_data()}.
+%% Module Entries (one per function, plus module header and footer).
+
+-type entry_data() :: term().
+
+-type tag() :: #tag{name :: atom(),
+		    line :: integer(),
+		    origin :: comment,
+		    data :: term()}.
+%% Generic tag information.
+
+-type edoc_module() :: xmerl_scan:xmlElement().
+%% The EDoc documentation data for a module,
+%% expressed as an XML document in {@link //xmerl. XMerL} format. See
+%% the file <a href="edoc.dtd">`edoc.dtd'</a> for details.
+
+-type ordset(T) :: ordsets:ordset(T).
+-type function_name() :: {atom(), integer()}.
+-type filename() :: file:filename().
+-type proplist() :: proplists:proplist().
+-type comment2() :: { Line :: integer(),
```

Comment:
Can we give this a better name than `comment2`?

Reply:
Definitely, just a second...

Reply:
I've pushed a rename. I've also found out that this type is actually an alias of `erl_comment_scan:comment()`, which is exported already. The data instances are read by `erl_comment_scan:read_comments()`. Should I mark it as `-type comment() :: erl_comment_scan:comment()`?

Reply:
yes, I think that would be better.